nautobot_ssot/integrations/vsphere/diffsync/adapters/adapter_vsphere.py

```diff
@@ -18,22 +18,28 @@
 )
 
 
-def deduce_network_from_ip(ip: ipaddress.IPv4Address, subnet_mask: str):
-    """Figure out the network of a given IP and its subnet mask."""
-    ip_int = int(ip)
-    subnet_int = int(ipaddress.IPv4Address(subnet_mask))
+def deduce_network_from_ip(ip: str, subnet_mask: str) -> str:
```

Comment:
Isn't this available through netutils? If it's not, it probably should be a function there.

Reply:
So, in digging deeper, it looks like that functionality is provided by IPAddress library. I refactored to use that instead.